refiner-lib/src/near_stream.rs

```diff
@@ -39,10 +53,23 @@ impl NearStream {
                 outcome.receipt.receiver_id.as_bytes() == self.context.engine_account_id.as_bytes()
             })
             .for_each(|outcome| {
-                self.handler.on_execution_outcome(near_block, outcome, &txs);
+                let rx_hash = &outcome.receipt.receipt_id;
+                let near_tx_hash = match self.tx_tracker.get_tx_hash(rx_hash) {
+                    Some(tx_hash) => tx_hash,
+                    None => {
+                        tracing::warn!("Transaction provenance unknown for receipt {}", rx_hash);
+                        Default::default()
```

Comment:
I'm just curious if the default value is ok here.

Reply:
This is a good question. I debated whether it makes sense to panic the refiner in this case, but I don't think we should. I think that if the transaction hash for a receipt is not known then it is better to flag it with monitoring and investigate rather than kill the whole process. This is because the bridge will only need a few transaction hashes so the probability of one missing causing a problem is very small, whereas crashing the process is guaranteed to create a headache for us.

In https://github.com/aurora-is-near/borealis-engine-lib/pull/31/commits/0a3a15ccfebb912cd0734cd51397700432b7abf1 I added a metric we can monitor so that we can catch problems if they do arise. I also made the new field optional for backwards compatibility reasons, and this means I can pass along `None` in this case instead of `default()`.